src/gf.c

```diff
@@ -2021,6 +2021,11 @@ JL_DLLEXPORT void jl_method_table_insert(jl_methtable_t *mt, jl_method_t *method
                         }
                         jl_array_ptr_1d_push(oldmi, (jl_value_t*)mi);
                         invalidate_external(mi, max_world);
+                        if (_jl_debug_method_invalidation) {
```

Comment:
I don't fully understand this code, but isn't this pushed into the list later if `invalidated` is set? Why do we need to push something here too? Is this supposed to have a loctag of `jl_box_int32(depth=1);` instead (and increment depth argument of invalidate_method_instance)?

Reply:
The method gets pushed, but not the specific MethodInstance. On previous versions, the format was:
```
backedge => depth (e.g., `mcc48954i::MethodInstance, 1` in the new test)
targetmi => "jl_method_table_insert" (`target = mc48954i::MethodInstance` in the new test)
targetm => "jl_method_table_insert" (`target = which(mc48954, (AbstractFloat, Int))` in the new test)
```
Recent master has dropped the middle of this pair; this PR just restores it.